src/shaka_scheme/system/gc/GC.cpp

```diff
@@ -0,0 +1,30 @@
+//
+// Created by Brandon Wong on 3/26/18.
+//
+
+#include "shaka_scheme/system/gc/GC.hpp"
+
+namespace shaka {
+
+    namespace gc {
+
+        GC::GC() {}
+        GC::~GC() {}
+
+        GC::GC(GC&& other) : list(std::move(other.list)) {}
+
+        GCData* GC::create_data(const shaka::Data& data) {
+            GCData *gcd = new GCData(data);
+            this->list.add_data(gcd);
+            return gcd;
+        }
+
+        int GC::get_size() {
+            return this->list.get_size();
+        }
+
+        void GC::sweep() {
+            this->list.sweep();
```

Comment:
Why is `sweep` a member function on `GCList`? I didn't realize this until now. Isn't the GC the entity that is supposed to sweep, and not the list? So the logic should exist at the GC level, not on the data container level with GCList.

Reply:
Well keeping in mind that GCList is a private member of GC, it really doesn't matter where the method is specified. Recall that GCList houses the private members of the GCData head pointer as well as the size of the list. This is why we placed `sweep()` inside of the GCList class, then further called it within GC.

Reply:
The reason we portend to define a sweep method on GCList is due to the fact that the list has purview over the actual internals of the list data structure, and can thus access the actual nodes to delete those that are unmarked, and re-establish connections as needed. A wrapper method will be added to the GC class that will delegate to this method, and be called from the top level.

Reply:
I feel like this violates the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) but I'll pass this.

Reply:
@VermillionAzure We can revisit this tomorrow during the Core team meeting and see if we can come up with a better approach for the next PR.